### ISSUE TITLE
Transaction request cannot get authorized from any accounts so it is blocked

### ISSUE THREAD

**Issue by paradigma-cl** [leather-io/extension#1377]
Trying to confirm a transaction from the Dapp, displays a modal window saying Unauthorized request blocking it. Followed the instructions of logging out, and log in again, but still does not considers this as a login.  Should the user delete the Accounts, and reinstall them to force asking for the password?  Probably, for each account should ask for the password after logout, but it will make it slower.  May be, it should ask for the password when there is a transaction request that considers the flow of funds.  
![imagen](https://user-images.githubusercontent.com/32182666/124205696-c68a0c80-daaf-11eb-8319-d978681e8a39.png)

**Comment by aulneau** [comment]
To clarify, are you logging in and out the the wallet, or the dapp?

**Comment by paradigma-cl** [comment]
Both.  I logged out the Dapp, and logged in the Dapp, many times, it did not work.
Then I logged out of the wallet, pressing the Stacks logo (it is not evident how to do this), and appears a model to send or receive in the account. The menu shows the way to log out.  The entered again the phrase + secretkey and tried to issue the transaction request.  It did not work again.

**Comment by aulneau** [comment]
What app are you using?

**Comment by paradigma-cl** [comment]
The Dapp is Paradigma CrossCheck https://CrossCheck.paradigma.global

Enviado desde mi iPhone

**Comment by paradigma-cl** [comment]
Looks like that this problem occurs when adding new accounts to the Web Wallet.  Initially, we have an account that authenticates correctly, has a Secret Key associated, and lets you execute contracts or transfers through the Stacks web wallet. Then after adding other new accounts, the first account loses the capacity to execute contracts or transfers giving the above message.  Then after deleting the added accounts, the functionality to execute is available again.

**Comment by paradigma-cl** [comment]
Just realized that all new accounts were generated from the same mnemonic.  This is very useful, but we have to see a way that the user realizes that is so.   The sequence of generation of those accounts is always the same?  Or one has to generate several in order to see which one is the used one?

**Comment by markmhendrickson** [comment]
It may be easiest to diagnose this issue over a call. I've been meaning to reach out and catch up about your progress with Paradigma CrossCheck in any case, so I'll send you a Discord now about that.

**Comment by paradigma-cl** [comment]
![imagen](https://user-images.githubusercontent.com/32182666/125495385-b1ee6534-c107-46a3-a54f-3bdef5df73ef.png)
The error with the console messages of the web wallet

**Comment by paradigma-cl** [comment]
@markmhx Thank you very much for the meeting yesterday!    I could not reproduce the 2nd error that we saw.

**Comment by markmhendrickson** [comment]
@paradigma-cl You're very welcome!

Is the console window in your [comment above](https://github.com/blockstack/stacks-wallet-web/issues/1377#issuecomment-879255387) the one that's shown when you right-click on the extension icon in your address bar and select "Inspect popup" specifically?

![image](https://user-images.githubusercontent.com/28991/125576802-14a958fa-5f47-4ef6-9fbb-6848a3fd98ab.png)

**Comment by paradigma-cl** [comment]
> 
> 
> @paradigma-cl You're very welcome!
> 
> Is the console window in your [comment above](https://github.com/blockstack/stacks-wallet-web/issues/1377#issuecomment-879255387) the one that's shown when you right-click on the extension icon in your address bar and select "Inspect popup" specifically?
> 
> ![image](https://user-images.githubusercontent.com/28991/125576802-14a958fa-5f47-4ef6-9fbb-6848a3fd98ab.png)

Yes, it is the same.  I used inspect when pressing the right button on top of the pop up window. And I used the Extensions Inspect Popup.  Both reach to the same message.

**Comment by paradigma-cl** [comment]
Now using the 2.12.2 Web Stacks Wallet
![imagen](https://user-images.githubusercontent.com/32182666/125833906-081719b0-d423-4999-8dd7-a43c15f3df31.png)

**Comment by aulneau** [comment]
@paradigma-cl would you be able to give me a very detailed rundown of how you're getting to this place?

The error is happening because for some reason the account you have signed into the app has been authenticated in a different way than what the extension is expecting. Does this app use the extension for authentication too?

**Comment by paradigma-cl** [comment]
> 
> 
> @paradigma-cl would you be able to give me a very detailed rundown of how you're getting to this place?
> 
> The error is happening because for some reason the account you have signed into the app has been authenticated in a different way than what the extension is expecting. Does this app use the extension for authentication too?

1. Initially from a browser without the Web Stacks Wallet, I access the dapp at https://crosscheck.paradigma.global
2. When pressing the button Secure Login with Stacks, it is required to download and install the Extension Web Stacks Wallet.
3. After, this is installed, then I chose an Existing account, so entered the words for the phrase, and the secret key.   This account has a Stacks domain name.
4. After, the browser opens the dapp.  Initially, shows some agreements as examples.
5. The way to use is to create a business agreement, like a Blank Document to start from scratch.  You enter the text of the purpose of the agreement in the space provided, create your own ID for the agreement and establish the dates of validity of it.  And press saves the document.
6. The agreement document will be created and will appear on the list of agreements.
7. When selecting it, you edit the agreement and can add some other information like annexes as attachments, pdf's files, other docs, images, videos, etc.  Supposedly to complement with specifications of the work to be done, etc.
8. In general, an agreement has an investment cost and a plan for how that is going to be paid.  That specification has to be entered in the Payment Form.   At the moment, the currency to pay is STX.
9. Now, it is time to define who is involved in the process of accepting the agreement and its role.  The users mentioned here will have access to the agreement document with the role assigned to them. The roles could be Authorizer (that accepts and sign the document), observer, payer (the one that pays if it is not the creator user), the payto (if it is different than the creator).  To specify the users, one has to use the Stacks naming system for the users.  If the user name is a valid Stacks name, will add it.  At least, requires 2 participants, and one with the role payto or payer. 
10. Then this document agreement is accessible by the other users using their account when opening Paradigma CrossCheck.
10. The agreement cannot be executed if the conditions are not accepted by signature.   So the participants with the role creator and authorizer have to sign it.
10.  When all the defined to authorize it pressed the Sign button, then this event can be registered in the Blockchain with the Register option. This registration freezes the possibility to change the agreement or the key participants of the agreement.
11. With this registration, the Payment option can be executed.  Depending on the payment form definition, there will be several lines where each one of them has a button to pay that specific line.
12. When pressing the button Pay button, a popup window will show the user the values associated with that line, and requires to press a button of Confirmation.  That button calls the Web Stacks Wallet using a contract call message.
13. The Web Stacks Wallet worked correctly up to this part, and subsequent lines.  Obviously, these are executed in different moments of the life of an agreement.

Now, this stopped working when some other Stacks account were added to the Web Stacks Wallet with the Generate button.  Well, that was what we discovered, in 3 different installations.
Applying the Switch option, also did not help.  So there is something, that makes the Web Stacks Wallet to "think" that it is not a valid call.
I tried to remove the generated accounts, to just leave one account only, but in Chrome under Windows, there was no option for it.

**Comment by markmhendrickson** [comment]
@paradigma-cl Is this issue still outstanding for you?

**Comment by paradigma-cl** [comment]
@markmhx Yes, we have tried different ways to try to detect the cause of the problem.  One test was to reduce the number of arguments to pass to the contract_call. All the ways persisted the same problem.
The second guess now is that as we have more than one account with the web wallet, probably the Stack's domain name is generating some issues.  As we are testing with the domains that were created with the older version of the Stacks ID (Auth), probably is causing the problem.  The first account that appears in the web wallet is the one with the .id.blockstack domain.  The following two (in my case) were generated from the words.  But these two do not have yet the subdomain name.  We have seen that the user name in the older Stacks ID (Auth) is stored in the profiles.name field.  But in the new structure, it is not in the profile.  I have not yet dig in to the Web Wallet code to see it.

**Comment by markmhendrickson** [comment]
@kyranjamie and @aulneau will join us next Wednesday during our monthly call to help troubleshoot this issue.

@paradigma-cl to confirm, you're not seeing this error when trying to sign transactions using the same extension install but another app (e.g. https://boom.money/) correct?

**Comment by paradigma-cl** [comment]
> 
> 
> @kyranjamie and @aulneau will join us next Wednesday during our monthly call to help troubleshoot this issue.
Welcome!
> 
> @paradigma-cl to confirm, you're not seeing this error when trying to sign transactions using the same extension install but another app (e.g. https://boom.money/) correct?

Yesterday I tried boom.money with the setup of 3 accounts, and it worked.  Still have the problem.

**Comment by paradigma-cl** [comment]
We have found the source of the problem.    In the call to a Clarity contract, the network argument variable was changed to a different value and not correctly set. So it changed the behaviour on the call to the web wallet.  The issue is solved.  Thank you very much for the effort and the help.
![imagen](https://user-images.githubusercontent.com/32182666/130523770-f385c92a-280b-40f8-8bfa-b3986fb581ea.png)
